selendroid-standalone/src/main/java/io/selendroid/standalone/server/model/DeviceStore.java

```diff
@@ -330,11 +330,17 @@ public boolean apply(AndroidDevice candidate) {
     return new Predicate<AndroidDevice>() {
       @Override
       public boolean apply(AndroidDevice candidate) {
+        // serial uniquely identifies a device, so we should immediately succeed
+        // if we find a match
+        String requestedSerial = capabilities.getSerial();
+        if (StringUtils.isNotBlank(requestedSerial)) {
+          return requestedSerial.equals(candidate.getSerial());
```

Comment:
Maybe it would be more useful to use equalsIgnoreCase. From my perspective it's a bit more robust and less error prone.

Reply:
ADB is case sensitive, so I think the code as written is correct.